### PR TITLE
Use context aware logger for workflow service

### PR DESCRIPTION
**Pull request by lkiesow**
The workflow service is using a special logger that supports units of work so that you can identify several logger statements belonging to a specific task.

However, with the new implementation, there is but one unit of work definition left, and it contains just a single logging statement. Nothing you actually still need this for.

```
logger.startUnitOfWork();
logger.warn(failureToFail, "Unable to update workflow to failed state");
logger.endUnitOfWork();
```

The downside to using this special logger is that you loose context information. With the normal logger you get information about the location where this has been logged. With the special logger this information is lost. That is really annoying when debugging based on logs:

```
2023-09-05 18:14:57,449 | INFO  | (ServiceRegistryJpaImpl:622) - Jobs with IDs '[8579450]' deleted
2023-09-05 18:14:57,450 | WARN  | (Log:213) - [>2ef22a08] No events for workflow instance 8579450 found in the Elasticsearch index.
2023-09-05 18:14:57,496 | DEBUG | (Log:213) - [>2ef22a08] Deleting workflow instance 8579614
2023-09-05 18:14:57,548 | INFO  | (Log:213) - [>2ef22a08] Removing temporary files for workflow 8579614
2023-09-05 18:14:57,548 | DEBUG | (Log:213) - [>2ef22a08] Removing temporary file https://oc-admin.virtuos.uni-osnabrueck.de/files/mediapackage/6a092362-56c4-46a9-b82d-f389d97ca19c/security-policy-series/xacml.xml for workflow Workflow {8579614}
```

Given that it is doing more harm than good, this patch switches the workflow service to the default logger. With this you also now get proper context information about logs:

```
2023-09-05 18:43:34,069 | INFO  | (ServiceRegistryJpaImpl:622) - Jobs with IDs '[9256488]' deleted
2023-09-05 18:43:34,073 | WARN  | (WorkflowServiceImpl:2260) - No events for workflow instance 9256488 found in the Elasticsearch index.
2023-09-05 18:43:34,082 | DEBUG | (WorkflowServiceImpl:2127) - Deleting workflow instance 9256489
2023-09-05 18:43:34,107 | INFO  | (WorkflowServiceImpl:881) - Removing temporary files for workflow 9256489
```

### Your pull request should…

* [x] have a concise title
* [x] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [x] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [x] include migration scripts and documentation, if appropriate
* [x] pass automated tests
* [x] have a clean commit history
* [x] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

### CODE REVIEW

**Review by Arnei — APPROVED** [review]
More context information is good, approved.

If me IDE isn't lying to me, this PR removes the last file where the `startUnitOfWork()` and `endUnitOfWork()` methods were actually used. Does this mean we can get rid of the special logger since its special logging capabilities are no longer required, or is there a reason to keep it around? Not that this PR should do the removing, but it would be worth an issue/follow-up PR.

**Comment by lkiesow** [comment]
I think we can and should remove it.
But that's another PR.
I'll create an issue for Hacktoberfest for that :D

File: modules/workflow-service-impl/src/main/java/org/opencastproject/workflow/impl/WorkflowServiceImpl.java [hunk 1]
```diff
@@ -69,7 +69,6 @@
 import org.opencastproject.serviceregistry.api.ServiceRegistry;
 import org.opencastproject.serviceregistry.api.ServiceRegistryException;
 import org.opencastproject.serviceregistry.api.UndispatchableJobException;
-import org.opencastproject.util.Log;
```
Comment (mtneug) [inline modules/workflow-service-impl/src/main/java/org/opencastproject/workflow/impl/WorkflowServiceImpl.java:72]: I haven't checked: Is this used anywhere else? Can this whole class be removed?

Comment (Arnei) [reply modules/workflow-service-impl/src/main/java/org/opencastproject/workflow/impl/WorkflowServiceImpl.java:72]: It's still used, so you can't "just" remove it. Shouldn't be too hard though. But as Lars said, that's another PR.

**Comment by lkiesow** [comment]
Filed https://github.com/opencast/opencast/issues/5261 about the removal of the logger

**Pull request merged**